### PR TITLE
update envoy: update interfaces, and fix broken macos ci

### PR DESCRIPTION
**Pull request by junr03**
Description: this submodule update brings in an updated macOS ci setup script that correctly install bazelisk. https://github.com/envoyproxy/envoy/pull/11444
Risk Level: low
Testing: CI

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

File: library/common/extensions/registry.cc [hunk 1]
```diff
@@ -14,6 +16,16 @@ void ExtensionRegistry::registerFactories() {
   Envoy::Extensions::StatSinks::MetricsService::forceRegisterMetricsServiceSinkFactory();
   Envoy::Extensions::TransportSockets::Tls::forceRegisterUpstreamSslSocketFactory();
   Envoy::Upstream::forceRegisterLogicalDnsClusterFactory();
+
+  // TODO: add a "force initialize" function to the upstream code, or clean up the upstream code
```
Comment (junr03) [inline library/common/extensions/registry.cc:20]: cc @goaway

File: test/integration/BUILD [hunk 2]
```diff
@@ -1,7 +1,7 @@
-licenses(["notice"])  # Apache 2
-
 load("@envoy//bazel:envoy_build_system.bzl", "envoy_cc_test", "envoy_package")
 
+licenses(["notice"])  # Apache 2
+
```
Comment (rebello95) [inline test/integration/BUILD:4]: Are these all deliberate?

Comment (junr03) [reply test/integration/BUILD:4]: Yep changed here https://github.com/envoyproxy/envoy/pull/11399

**Comment by buildbreaker** [comment]
`android / linux_dist (pull_request)` keeps getting canceled for some reason for my PRs too :(

**Comment by junr03** [comment]
looks like the android linux_dist job ran out of time https://github.com/lyft/envoy-mobile/pull/879/checks?check_run_id=743616566

going to merge this and check what is up with the time limit

**Pull request merged**